batch/batch/front_end/front_end.py

```diff
@@ -718,7 +720,7 @@ def assert_is_sha_1_hex_string(revision: str):
         raise web.HTTPBadRequest(reason=f'revision must be 40 character hexadecimal encoded SHA-1, got: {revision}')
 
 
-async def _create_jobs(userdata: dict, job_specs: dict, batch_id: int, update_id: int, app: aiohttp.web.Application):
+async def _create_jobs(userdata, job_specs: dict, batch_id: int, update_id: int, app: web.Application):
```

Comment:
Good catch on `job_specs`. Hrm, `userdata` is trickier here. Looks like `_create_jobs` intentionally reassigns `userdata` to a more limited dictionary to avoid propagating any sensitive information into the specs. I feel like 1. the reassigning to a different type is bad, but also I feel like there shouldn't be sensitive information in the `userdata` in the first place? I don't see why `auth` should send any info that isn't public back to other services. Can we address this issue in a follow-up instead of this PR?